sdk/core/azure-core/azure/core/paging_method.py

```diff
@@ -0,0 +1,225 @@
+        """
+        raise NotImplementedError("This method needs to be implemented")
+
+    def mutate_list(self, pipeline_response, list_of_elem):
+        # type: (HttpResponse, Iterable[ReturnType]) -> Iterable[ReturnType]
+        """Mutate list of elements in current page, i.e. if users passed in a cls calback
+        """
+        raise NotImplementedError("This method needs to be implemented")
+
+    def get_continuation_token(self, pipeline_response, deserialized):
+        # type: (HttpResponse, ResponseType) -> Any
+        """Get the continuation token from the current page
+        """
+        raise NotImplementedError("This method needs to be implemented")
+
+    def extract_data(self, pipeline_response):
+        # type: (HttpResponse) -> Tuple[Any, Iterable[ReturnType]]
+        """Return the continuation token and current list of elements to PageIterator
+        """
+        raise NotImplementedError("This method needs to be implemented")
+
+
+class BasicPagingMethod(PagingMethodABC):
+    """This is the most common paging method. It uses the continuation token
+    as the next link
+    """
+
+    def __init__(self):
+        self._client = None
+        self._deserialize_output = None
+        self._path_format_arguments = None
+        self._item_name = None
+        self._next_link_name = None
+        self.did_a_call_already = False
+        self._cls = None
+        self._error_map = None
+
+    def initialize(self, client, deserialize_output, next_link_name, **kwargs):
+        # type: (PipelineClient, Callable, str, Any) -> None
+        self._client = client
+        self._deserialize_output = deserialize_output
+        self._next_link_name = next_link_name
+
+        self._path_format_arguments = kwargs.pop("path_format_arguments", {})
+        self._item_name = kwargs.pop("item_name", "value")
+        self._cls = kwargs.pop("_cls", None)
+
+        self._error_map = {
+            401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError
+        }
+        self._error_map.update(kwargs.pop('error_map', {}))
+
+    def get_next_request(self, continuation_token, initial_request):
+        # type: (Any, HttpRequest) -> HttpRequest
+        next_link = continuation_token
+        next_link = self._client.format_url(next_link, **self._path_format_arguments)
+        request = initial_request
+        request.url = next_link
+
+        return request
+
+    def get_page(self, continuation_token, initial_request):
+        # type: (Any, HttpRequest) -> HttpResponse
+        if not self.did_a_call_already:
+            request = initial_request
+            self.did_a_call_already = True
+        else:
+            request = self.get_next_request(continuation_token, initial_request)
+        response = self._client._pipeline.run(request, stream=False)  # pylint: disable=protected-access
+
+        http_response = response.http_response
+        if not 200 <= http_response.status_code < 300:
```

Comment:
Nitpick: reading doing `status_code < 200 or status_code >= 300` is marginally easier to read. At least that is what I have to mentally translate the line of code to say. But maybe that is my Swedish accent coming into play.

Reply:
sounds good, I def see where you're coming from